suite-common/wallet-core/src/transactions/transactionsReducer.ts

```diff
@@ -200,16 +202,37 @@ export const selectTransactionFirstTargetAddress = memoizeWithArgs(
     },
 );
 
-export const selectTransactionFirstInputAddress = memoizeWithArgs(
+const selectTransactionTargetAddresses = memoizeWithArgs(
     (state: TransactionsRootState, txid: string, accountKey: AccountKey) => {
+        const transactionTargets = selectTransactionTargets(state, txid, accountKey);
+        if (G.isNullable(transactionTargets)) return [];
+
+        return mapTransactionInputsOutputsToAddresses(transactionTargets);
+    },
+);
+
+export const selectTransactionInputAddresses = memoizeWithArgs(
+    (state: TransactionsRootState, txid: string, accountKey: AccountKey): string[] => {
         const transaction = selectTransactionByTxidAndAccountKey(state, txid, accountKey);
-        return transaction?.details?.vin?.[0].addresses?.[0];
+
+        if (G.isNullable(transaction)) return [];
+
+        const inputAddresses = mapTransactionInputsOutputsToAddresses(transaction.details.vin);
+        const targetAddresses = selectTransactionTargetAddresses(state, txid, accountKey);
+
+        return sortTargetAddressesToBeginning(inputAddresses, targetAddresses);
```

Comment:
nit: if you want, this would be a nice use case for pipe 😉

Reply:
How? There is basically nothing happening in these selectors. How would you pipe it? Or am I missing something?